include/rcpputils/filesystem_helper.hpp

```diff
@@ -0,0 +1,195 @@
+// are met:
+//
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above
+//    copyright notice, this list of conditions and the following
+//    disclaimer in the documentation and/or other materials provided
+//    with the distribution.
+//  * Neither the name of the copyright holders nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// This file is originally from:
+// https://github.com/ros/pluginlib/blob/1a4de29fa55173e9b897ca8ff57ebc88c047e0b3/pluginlib/include/pluginlib/impl/filesystem_helper.hpp
+
+/// Includes std::filesystem and aliases the namespace to `pluginlib::impl::fs`.
+/**
+ * If std::filesystem is not available the necessary functions are emulated.
+ */
+
+#ifndef RCPPUTILS__FILESYSTEM_HELPER_HPP_
+#define RCPPUTILS__FILESYSTEM_HELPER_HPP_
+
+#if defined(_MSC_VER)
+# if _MSC_VER >= 1900
+#  include <experimental/filesystem>
+namespace rcpputils
+{
+namespace fs = std::experimental::filesystem;
+}  // namespace rcpputils
+
+#  define RCPPUTILS__FILESYSYEM_HELPER__HAS_STD_FILESYSTEM
+# endif
+
+#elif defined(__has_include)
+# if __has_include(< filesystem >) && __cplusplus >= 201703L
+#  include <filesystem>
+
+namespace rcpputils
+{
+namespace fs = std::filesystem;
+}  // namespace rcpputils
+
+#  define RCPPUTILS__FILESYSYEM_HELPER__HAS_STD_FILESYSTEM
+// cppcheck-suppress preprocessorErrorDirective
+# elif __has_include(< experimental / filesystem >)
+#  include <experimental/filesystem>
+
+namespace rcpputils
+{
+namespace fs = std::experimental::filesystem;
+}  // namespace rcpputils
+
+#  define RCPPUTILS__FILESYSYEM_HELPER__HAS_STD_FILESYSTEM
+# endif
+#endif
+
+// The standard library does not provide it, so emulate it.
+#ifndef RCPPUTILS__FILESYSYEM_HELPER__HAS_STD_FILESYSTEM
+
+#include <string>
+#include <vector>
+
+#ifdef _WIN32
+#define RCPPUTILS_IMPL_OS_DIRSEP '\\'
+#else
+#define RCPPUTILS_IMPL_OS_DIRSEP '/'
+#endif
+
+#ifdef _WIN32
+  #include <io.h>
```

Comment:
Here and a few other places are not indented properly according to https://google.github.io/styleguide/cppguide.html#Preprocessor_Directives 
```
The hash mark that starts a preprocessor directive should always be at the beginning of the line.
```


It's also not consistent with the style of the rest of the file, i.e.

```
#ifdef FOO
# define BAR
#endif
```